oauth_provider/tests/test_oauth_provider_client.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 SYLEAM
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from openerp.tests.common import TransactionCase
+from ..oauth2.validator import OdooValidator
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from oauthlib import oauth2
+except ImportError:
+    _logger.debug('Cannot `import oauthlib`.')
+
+
+class TestOAuthProviderClient(TransactionCase):
+
+    def setUp(self):
+        super(TestOAuthProviderClient, self).setUp()
+        self.client_vals = {
+            'name': 'Client',
+            'identifier': 'client',
+        }
+
+    def new_client(self, vals=None):
+        values = self.client_vals.copy()
+        if vals is not None:
+            values.update(vals)
+
+        return self.env['oauth.provider.client'].create(values)
+
+    def test_grant_response_type_fields(self):
+        """
+        Check the value of the grant_type and response_type fields
+        """
+        # Default : Web Application
+        client = self.new_client({'identifier': 'default'})
+        self.assertEqual(client.grant_type, 'authorization_code')
+        self.assertEqual(client.response_type, 'code')
+
+        # Web Application
```

Comment:
Yeah the smaller the test the better. I go to great lengths to follow a 1:1 method:assertion ratio, but sometimes it's not possible given context and setup required. 

Consider an example using these tests. If the response_type for the web application is wrong, we won't know if it's right for Mobile, Legacy, and Backend because the test will stop execution at the first failed assertion. 

It's incredibly valuable to have the whole picture when debugging failing tests because it can help point us closer to the actual error, or specific edge case that is being debugged.